### PR TITLE
Bug no duplicates

### LINKED ISSUE #9: Bug: axe runner results missing errors
The axe runner results are missing errors in some cases.  It appears that `axe-core` is grouping errors by type in the `violations` array, and each applicable element is a `target` within that violations' `nodes` array.  But, the runner is only taking the first value, so the results only include the first of each type.

The attached file [pa11y-axe-runner.zip](https://github.com/pa11y/pa11y-runner-axe/files/3990550/pa11y-axe-runner.zip) has example reports from `pa11y` with the axe runner and `axe-core` run against https://www.w3.org/WAI/demos/bad/before/news.html as examples (which includes multiple errors for some types).

The issue appears to be in the code below (noted as needing more testing), where `selector` can contain multiple comma delimited values, but `element` only includes the first:

```js
function processIssue(issue) {
	// For now we just select the first element. This needs testing
	const selector = issue.nodes.map(node => selectorToString(node.target)).join(', ');
	const element = (issue.nodes.length ? window.document.querySelector(selector) : null);
```

This fix has not had a lot of testing, but appeared to work as at least a proof of concept:

```js
async function runAxeCore() {
	const result = await window.axe.run();
	return [].concat(
		result.violations.map(processViolation).flat(),
		result.incomplete.map(processIncomplete).flat()
	);
}

function processIssue(issue) {
	let elements = [ null ];
	if (issue.nodes.length) {
		const selectors = issue.nodes.map(node => selectorToString(node.target));
		elements = selectors.map(selector => window.document.querySelector(selector));
	}
	
	return elements.map(element => ({
		code: issue.id,
		message: `${issue.help} (${issue.helpUrl})`,
		type: issue.type,
		element,
		runnerExtras: {
			description: issue.description,
			impact: issue.impact,
			help: issue.help,
			helpUrl: issue.helpUrl
		}
	}));
}
```

The same tests run with this code are the `-fixed` files in the zip.

### PR DESCRIPTION
**Pull request by joeyciechanowicz**
Fixes #9 
Theres a duplicate issue on the pa11y repo https://github.com/pa11y/pa11y/issues/506

aXe reports one instance of an error for each error. inside the aXe object theres an array of nodes. this PR flattens that array into multiple separate issues.

### CODE REVIEW

File: lib/runner.js [hunk 1]
```diff
@@ -72,27 +72,31 @@ runner.run = async () => {
...
 	 */
-	function processIssue(issue) {
+	function processIssue(axeIssue) {
+		// See https://github.com/dequelabs/axe-core/blob/develop/doc/API.md#results-object for a description
+		// of an axe issue
 
-		// For now we just select the first element. This needs testing
-		const selector = issue.nodes.map(node => selectorToString(node.target)).join(', ');
-		const element = (issue.nodes.length ? window.document.querySelector(selector) : null);
+		let elements = [null];
```
Comment (kkoskelin) [inline lib/runner.js:82]: How's test coverage when `axeIssues.nodes.length` is zero? Will this do what we want?

Comment (joeyciechanowicz) [reply lib/runner.js:82]: There is an entry in the test with `id: 'mock-id-no-nodes'` for this line.

Comment (joeyciechanowicz) [reply lib/runner.js:82]: NYC is reporting 100% coverage

**Review by hollsk — APPROVED** [review]
✅

File: lib/runner.js [hunk 2]
```diff
@@ -44,7 +44,7 @@ runner.run = async () => {
 		return [].concat(
 			result.violations.map(processViolation),
 			result.incomplete.map(processIncomplete)
-		);
+		).reduce((acc, curr) => acc.concat(curr), []);
```
Comment (kkoskelin) [inline lib/runner.js]: I frequently lean on things like lodash to `flatten(arr1, arr2)` as the concat/reduce construction is difficult to read (IMO) and 100% coverage of the reducer function itself is unnecessarily difficult. But I think I'm getting the idea that we're trying to keep external modules to a minimum. Nice that this new code IS covered.

**Review by kkoskelin — APPROVED** [review]

File: lib/runner.js (hunk 2 above)
Comment (joeyciechanowicz) [reply lib/runner.js]: I'm not keen to add more dependencies for something so simple, but unless you're familiar then the `.reduce` does take time to get your mind round.
I've removed it in favour of `...spread` which is a lot more readable IMO

**Review by kkoskelin — APPROVED** [review]
💯

**Pull request merged**